src/olympia/users/admin.py

```diff
@@ -190,21 +186,33 @@ def get_search_results(self, request, queryset, search_term):
         ips = self.ip_addresses_if_query_is_all_ip_addresses(search_term)
         if ips:
             q_objects = Q()
-            annotations = {}
-            for arg in self.extra_list_display_for_ip_searches:
-                q_objects |= Q(**{f'{arg}__in': ips})
-                if '__' in arg:
-                    annotations[arg] = F(arg)
-            queryset = queryset.filter(q_objects).annotate(**annotations)
-            # We force the distinct() ourselves and tell Django there are no
-            # duplicates, otherwise the admin de-duplication logic, which
-            # doesn't use distinct() after Django 3.1, would break our
-            # annotations.
-            # This can cause some users to show up multiple times, but that's
-            # a feature: it will happen when the IPs returned are different
-            # (so technically the rows are not duplicates), since the
-            # annotations are part of the distinct().
-            queryset = queryset.distinct()
+            for ip in ips:
+                q_objects |= Q(
+                    activitylog__iplog__ip_address_binary=ipaddress.ip_address(
+                        ip
+                    ).packed
+                )
```

Comment:
I'd natively assume this could be replace with a single `__in` query:
`q_objects = Q(activitylog__iplog__ip_address_binary__in=[ipaddress.ip_address(ip).packed for ip in ips])`
or does that not work?

Reply:
I naively assumed it wouldn't work because it's all binary strings, but actually, you're probably right, it might work...